### PR TITLE
Allow changing debug settings via Lua script

### PR DESCRIPTION
**Pull request by maxim-productengine**
Add the way to change debug settings via Lua script. This change won't persist between sessions.
Also made minor adjustment to 'Debug settings' floater to allow copy the name of the setting to the clipboard.

### CODE REVIEW

**Review by nat-goodspeed — CHANGES_REQUESTED** [review]
I think Lua access to settings should target [`LLViewerControlListener`](https://github.com/secondlife/viewer/blob/release/luau-scripting/indra/newview/llviewercontrollistener.cpp) instead of adding `set|getDebugSetting()` to `LLAppViewerListener`. 

Even if you do add these entry points to `LLAppViewerListener`, `LLViewerControlListener` should still pass the new `saved_value=false` parameter.

I don't mind if the Lua module is still called `LLDebugSettings`.

File: indra/newview/llappviewerlistener.cpp [hunk 1]
```diff
@@ -48,6 +49,14 @@ LLAppViewerListener::LLAppViewerListener(const LLAppViewerGetter& getter):
     add("forceQuit",
         "Quit abruptly",
         &LLAppViewerListener::forceQuit);
+
+    add("setDebugSetting", 
+        "Apply specified [\"value\"] to the debug [\"setting\"] (this change won't persist across sessions)",
```
[the remark this answers is not in the document]
Comment (maxim-productengine) [reply indra/newview/llappviewerlistener.cpp:54]: Hm, I like it, thanks.

File: indra/newview/scripts/lua/LLDebugSettings.lua [hunk 2]
```diff
@@ -0,0 +1,13 @@
+leap = require 'leap'
+
+local LLDebugSettings = {}
+
+function LLDebugSettings.set(name, value)
+    leap.send('LLAppViewer', {op='setDebugSetting', setting=name, value=value})
```
[the remark this answers is not in the document]
Comment (maxim-productengine) [reply indra/newview/scripts/lua/LLDebugSettings.lua]: Agree, seems reasonable.

File: indra/newview/scripts/lua/LLDebugSettings.lua [hunk 3]
```diff
@@ -0,0 +1,17 @@
+leap = require 'leap'
+
+local LLDebugSettings = {}
+
+function LLDebugSettings.set(name, value)
+    return leap.request('LLViewerControl', {op='set', group='Global', key=name, value=value})
```
Comment (nat-goodspeed) [inline indra/newview/scripts/lua/LLDebugSettings.lua:6]: Should `leap.request()` and `leap.generate()` notice an `"error"` key in the returned data and raise the specified error?

If they don't, maybe we should do it here and in `toggle()`.

File: indra/newview/llappviewerlistener.h [hunk 4]
```diff
@@ -47,6 +47,8 @@ class LLAppViewerListener: public LLEventAPI
     void requestQuit(const LLSD& event);
     void forceQuit(const LLSD& event);
 
+    void setDebugSetting(const LLSD &event);
+
```
Comment (nat-goodspeed) [inline indra/newview/llappviewerlistener.h:51]: Seems like you could revert the changes to both `llappviewerlistener.{h,cpp}`.

File: indra/newview/llviewercontrollistener.cpp [hunk 5]
```diff
@@ -158,7 +159,8 @@ void LLViewerControlListener::toggle(LLSD const & request)
 
 	if (info.control->isType(TYPE_BOOLEAN))
 	{
-		info.control->set(! info.control->get().asBoolean());
+        LL_WARNS("LLViewerControlListener") << "Toggling debug setting " << std::quoted(info.key) << LL_ENDL;
+		info.control->set(! info.control->get().asBoolean(), false);
```
Comment (nat-goodspeed) [inline indra/newview/llviewercontrollistener.cpp:163]: Let's capture `! info.control->get().asBoolean()` as a local variable first so we can report it in the `LL_WARNS()` before setting it.

**Review by nat-goodspeed — CHANGES_REQUESTED** [review]

File: indra/newview/scripts/lua/LLDebugSettings.lua [hunk 6]
```diff
@@ -0,0 +1,24 @@
+leap = require 'leap'
+
+function check_response(res)
```
Comment (nat-goodspeed) [inline indra/newview/scripts/lua/LLDebugSettings.lua:3]: Let's make this a `local` function. I like it. Does it work when you pass an unknown group or control name?

**Review by nat-goodspeed — APPROVED** [review]

File: indra/newview/scripts/lua/LLDebugSettings.lua (hunk 6 above)
Comment (maxim-productengine) [reply indra/newview/scripts/lua/LLDebugSettings.lua:3]: Yes, I've checked it.
